src/System.Runtime.Extensions/tests/System/EnvironmentTests.cs

```diff
@@ -215,6 +215,25 @@ public void GetFolderPath_Unix_PersonalIsHomeAndUserProfile()
             Assert.Equal(Environment.GetEnvironmentVariable("HOME"), Environment.GetFolderPath(Environment.SpecialFolder.UserProfile));
         }
 
+        [Theory]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]  // Tests OS-specific environment
```

Comment:
All of the below InlineData passed locally. After the build is finished I intend to reduce the following InlineData. Because in order to exercise the code path that this PR fixes one of the following InlineData would be sufficient.

Reply:
I don't care whether you have all those variations - in fact that makes it more sure that at least one will exercise the "create" path.

As you can see the logic to choose tose locations is complex, it is not clear they are all missing on all distros, and in fact your test results prove they arent:
https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Extensions/src/System/Environment.Unix.cs#L95